### DISCUSSION TITLE
MPPT Algorithm

### DISCUSSION THREAD

**Discussion by mrkasri** [pvlib/pvlib-python#1816] [Q&A]
Do the inverters implement MPPT algorithms? If not the case how to implement p & o algorithm using pvlib ?

**Comment by cwhanse** [comment]
Inverters have MPPT algorithm, but pvlib-python doesn't have those functions. The inverter functions in pvlib are DC-AC conversion efficiency models.

**Comment by mrkasri** [comment]
Is there any way to add these functions (mppt algorithm) in the code.

**Reply by cwhanse** [reply]
I don't think an MPPT simulation would fit in pvlib. A MPPT simulator would need to operate on a faster time scale (e.g., 1 kHz) than the other models in the library are designed for. For example, the DC power models in pvlib do not explicitly account for fast time scale processes such as module capacitance. If you are attempting to simulate the MPPT to explore improvements in MPPT efficiency, I would like a circuit simulator like SPICE would be a better choice.

**Comment by mrkasri** [comment]
Thank you Mr Hansen. I'll give it a try.
